akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -352,7 +494,8 @@ final class GraphInterpreterShell(
    *  - if the event limit is greater than maxInputBufferSize * (ins + outs) than there will always be expand activity
    *  because no data can enter “fast enough” from the outside
    */
-  val shellEventLimit = settings.maxInputBufferSize * (assembly.ins.length + assembly.outs.length)
+  // TODO: Fix event limit heuristic
+  val shellEventLimit = settings.maxInputBufferSize * 16
```

Comment:
I guess the number of internal connections is a good multiplier (i.e. length of the connections array)